shell/platform/darwin/ios/framework/Source/FlutterEngineGroup.mm

```diff
@@ -51,27 +51,26 @@ - (FlutterEngine*)makeEngineWithOptions:(nullable FlutterEngineGroupOptions*)opt
   NSArray<NSString*>* entrypointArgs = options.entrypointArgs;
 
   FlutterEngine* engine;
+  // NSPointerArray is clever and assumes that unless a mutation operation has occurred on it that
+  // has set one of its values to nil, nothing could have changed and it can skip compaction.
+  // That's reasonable behaviour on a regular NSPointerArray but not for a weakObjectPointerArray.
+  // As a workaround, we mutate it first. See: http://www.openradar.me/15396578
+  [self.engines addPointer:nil];
+  [self.engines compact];
```

Comment:
I won't comment on just how much time I spent staring at and debugging this line to arrive at the comment above. Thanks, noble open radar submitter, for saving my sanity.

Reply:
😭